tests/brokers_sync/base/connection.py

```diff
@@ -0,0 +1,52 @@
+from typing import Type
+
+import pytest
+
+from propan.brokers._model import BrokerSyncUsecase
+
+
+class BrokerConnectionTestcase:
+    broker: BrokerSyncUsecase
+
+    def get_broker_args(self, settings):
+        return (settings.url,), {}
+
+    def ping(self, broker) -> bool:
+        return True
+
+    def test_warning(self, broker: BrokerSyncUsecase):
+        with broker:
+            broker.start()
+            assert broker.started
+            with pytest.warns(RuntimeWarning):
+                broker.handle("test")
+        assert not broker.started
+
+
+    def test_init_connect_by_url(self, settings):
+        args, kwargs = self.get_broker_args(settings)
+        broker = self.broker(*args, **kwargs)
+        assert broker.connect()
```

Comment:
Can you please refactor all the following testcases to use a mocked `_connect` method instead? We can check incoming `*args, **kwargs` arguments by `assert_called_with` method, so we need no more connect to the real broker to test correct arguments merging.

Reply:
I thought about replacing the original broker `_connect` method to check incoming arguments. It will be suitable for all brokers, not RMQ only

Reply:
Can you give me an example?

Reply:
Tomorrow

Reply:
By words: you need to patch this one https://github.com/Lancetnik/Propan/blob/main/propan/brokers/rabbit/rabbit_broker.py#L108 the way you patching in the TestBroker
'broker._connect = MethodType(async_mock, broker)' and check the mock call arguments